lib/1.4/dml-builtins.dml

```diff
@@ -997,6 +999,48 @@ template init_as_subobj is (connect, init, _qname) {
     }
 }
 
+/**
+### map\_target
+
+A `connect` object can instantiate the template `map_target`, making it only
+accept objects that are valid map targets, i.e. objects for which
+`SIM_new_map_target` returns a valid pointer, without exception. The template
+provides a session variable `map_target` of type `map_target_t *` which can be
+used to issue transactions to the connected object. The template also defines a
+method `issue(transaction_t *t, uint64 addr) -> (exception_type_t)` which can
+be used as a shorthand for `SIM_issue_transaction`.
+*/
+template map_target is connect {
```

Comment:
This template is highly Simics-specific, both in name and in signatures. I want to keep dml-builtins as simulator agnostic as possible.

I propose three things:
* move to utility.dml
* add alternative shorthand simulator-agnostic methods along the lines of `memory_space.{read,write}`
* Rename to something more simulator-agnostic, like `memory`.

I have no problem keeping the `issue` method; we can view it as a simulator-specific extension.

Reply:
> move to utility.dml

Sure, I put it in builtins because it felt similar to `init_as_subobj` and I wanted them documented together.

> alternative shorthand simulator-agnostic

Ok, that could probably be useful

> rename to something more simulator-agnostic

I want to keep it agnostic with respect to what it targets too, so I don't really like `memory` since it could just as well target a register-bank. I think map_target _could_ be simulator agnostic, we we define it as such.

Reply:
Off-topic: 
Maybe init_as_subj should be moved to utility as well? It is not done since we need to be able send params to the subobj and that may become too Simics specific as well?

Reply:
I agree that init_as_subobj logically would belong to utility; the reason I chose dml-builtins is that I needed some compiler hacks to make it work (because we cannot yet call methods from module init). Things that are tightly coupled with the compiler count as builtins.

Reply:
Ok, so I'll put it in utility.dml, how do we get it documented together with `init_as_subobj`?

Reply:
Through hyperlinks.